gcloud-java-resourcemanager/src/main/java/com/google/gcloud/resourcemanager/testing/LocalResourceManagerHelper.java

```diff
@@ -353,16 +353,27 @@ Response get(String projectId, String[] fields) {
   }
 
   Response list(Map<String, Object> options) {
-    // Use pageSize and pageToken options when Cloud Resource Manager does so (#421)
     List<String> projectsSerialized = new ArrayList<>();
     String[] filters = (String[]) options.get("filter");
     if (filters != null && !isValidFilter(filters)) {
       return Error.INVALID_ARGUMENT.response("Could not parse the filter.");
     }
     String[] fields = (String[]) options.get("fields");
+    int count = 0;
+    String pageToken = (String) options.get("pageToken");
+    Integer pageSize = (Integer) options.get("pageSize");
+    String nextPageToken = null;
     for (Project p : projects.values()) {
+      if (pageToken != null && p.getProjectId().compareTo(pageToken) < 0) {
```

Comment:
Because the keys are projectId you can use [tailMap](https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ConcurrentSkipListMap.html#tailMap(K)).